### PR TITLE
fetch_guild, with with_counts

### PR DESCRIPTION
**Pull request by JDJGInc**

### CODE REVIEW

**Comment by JDJGInc** [comment]
image

**Comment by JDJGInc** [comment]
image
tested it too

**Comment by JDJGInc** [comment]
tested with:
from disnake.ext import commands
import disnake
import typing
from disnake.ext.commands.core import command

bot = commands.Bot(command_prefix="e*", intents=disnake.Intents.all())


@bot.event
async def on_ready():
    print(f"Bot is ready")


@bot.command(brief="gives information on snowflakes")
async def snowflake_info(ctx, *, snowflake: typing.Optional[disnake.Object] = None):

    if not snowflake:
      await ctx.send("you either returned nothing or an invalid snowflake now going to the current time for information.")

    generated_time = await commands.ObjectConverter().convert(ctx, argument=f"{int((disnake.utils.utcnow()).timestamp() * 1000 - 1420070400000) << 22 | 0x3fffff}")
    #yes the later bit uses generate_snowflake's older version

    snowflake = snowflake or generated_time

    embed = disnake.Embed(title="❄️ SnowFlake Info:", color=5793266)

    embed.add_field(name="Created At:",
                    value=f"{disnake.utils.format_dt(snowflake.created_at, style = 'd')}\n{disnake.utils.format_dt(snowflake.created_at, style = 'T')}")

    embed.add_field(name="Worker ID:",
                    value=f"{snowflake.worker_id}")

    embed.add_field(name="Process ID:",
                    value=f"{snowflake.process_id}")

    embed.add_field(name="Increment:", value=f"{snowflake.increment_id}")

    embed.set_footer(text=f"Snowflake ID: {snowflake.id}")

    await ctx.send(embed=embed)

TOKEN = ""
bot.run(TOKEN)

**Comment by thesadru** [comment]
I don't think this is actually needed in any way? I'm all for adding a utility function to parse the snowflake but I don't like cluttering the `disnake.Object` with attributes few people would use. At the very least the utility function should be a priority and properties in the class an after-thought.

Although on the other hand it was also thought that `parse_token` won't really have that much of a use-case and now it's used surprisingly often.

**Comment by JDJGInc** [comment]
> I don't think this is actually needed in any way? I'm all for adding a utility function to parse the snowflake but I don't like cluttering the `disnake.Object` with attributes few people would use. At the very least the utility function should be a priority and properties in the class an after-thought.
> 
> Although on the other hand it was also thought that `parse_token` won't really have that much of a use-case and now it's used surprisingly often.

>___> a lot of people would probaly having a snowflake info command tbh.

**Comment by JDJGInc** [comment]
> I don't think this is actually needed in any way? I'm all for adding a utility function to parse the snowflake but I don't like cluttering the `disnake.Object` with attributes few people would use. At the very least the utility function should be a priority and properties in the class an after-thought.
> 
> Although on the other hand it was also thought that `parse_token` won't really have that much of a use-case and now it's used surprisingly often.

I accidently merged my two pull requests into one .___.
 but uh

**Comment by JDJGInc** [comment]
if people don't want an object update, I can just pass the new fetch_guild that gets the counts?

**Comment by thesadru** [comment]
Please keep this in a separate pull request. These two changes are very different and shouldn't be grouped together.

**Comment by JDJGInc** [comment]
> Please keep this in a separate pull request. These two changes are very different and shouldn't be grouped together.

IT was an issue before,  I will slipt them

**Comment by JDJGInc** [comment]
> Please keep this in a separate pull request. These two changes are very different and shouldn't be grouped together.

weird bugs.

**Comment by JDJGInc** [comment]
image

**Comment by JDJGInc** [comment]
image
Also

**Comment by JDJGInc** [comment]
import disnake
client = disnake.Client(intents=disnake.Intents.all())

guild_id = #empty but you can pass an int for this.

@client.event
async def on_ready():
    print("client is ready")

    guild = await client.fetch_guild(guild_id)
    print(guild.approximate_member_count, guild.approximate_presence_count)

    guild = await client.fetch_guild(guild_id, with_counts=False)
    print(guild.approximate_member_count, guild.approximate_presence_count)

TOKEN = ""
#you can pass the TOKEN here
client.run(TOKEN)

**Comment by JDJGInc** [comment]
https://mystb.in/AncientGrandLeaf.python

**Comment by JDJGInc** [comment]
output:
client is ready
523 302
None None

**Comment by JDJGInc** [comment]
I do apoligize for swap around the pull requests, it was the easy fix for me, however it should work well, with this anyway so

File: disnake/guild.py [hunk 1]
```diff
@@ -241,7 +241,18 @@ class Guild(Hashable):
...
         The guild's NSFW level.
 
-        .. versionadded:: 2.0
+    approximate_member_count: Optional[:class:`int`]
+        The approximate number of members in the guild. This is ``None`` unless the guild is obtained
+        using :meth:`Client.fetch_guild` with ``with_counts=True``.
+        
+        
+        
+    approximate_presence_count: Optional[:class:`int`]
```
Comment (Victorsitou) [inline disnake/guild.py:250]: Please avoid unnecessary spaces.

**Comment by thesadru** [comment]
Closing as the quality of this pr is unmanageable.

**Pull request closed without merging**